addons/project/models/project.py

```diff
@@ -244,23 +244,22 @@ def _compute_attached_docs_count(self):
     def _compute_task_count(self):
         domain = [('project_id', 'in', self.ids), ('is_closed', '=', False)]
         fields = ['project_id', 'display_project_id:count']
-        groupby = ['project_id']
-        task_data = self.env['project.task']._read_group(domain, fields, groupby)
+        groupby = ['project_id', 'active']
         result_wo_subtask = defaultdict(int)
         result_with_subtasks = defaultdict(int)
-        for data in task_data:
-            result_wo_subtask[data['project_id'][0]] += data['display_project_id']
-            result_with_subtasks[data['project_id'][0]] += data['project_id_count']
-        task_all_data = self.env['project.task'].with_context(active_test=False)._read_group(domain, fields, groupby)
-        all_tasks_wo_subtasks = defaultdict(int)
+        task_all_data = self.env['project.task'].with_context(active_test=False)._read_group(domain, fields, groupby, lazy=False)
         for data in task_all_data:
-            all_tasks_wo_subtasks[data['project_id'][0]] += data['display_project_id']
+            project_id = data['project_id'][0]
+            if data['active'] or not self.browse(project_id).active:
```

Comment:
Instead of doing a browse for each project, we could store in a variable the active projects by filtering the projects contained in `self` and then checking if the project_id is in that variable. 🤔 🙂 

```py
        active_project_ids = self.filtered('active').ids
        for data in task_all_data:
            project_id = data['project_id'][0]
            if data['active'] or project_id not in active_project_ids:
```